src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -263,16 +301,27 @@ public String toJSON(MaxwellOutputConfig outputConfig) throws IOException {
 			}
 		}
 
-		writeMapToJSON("data", this.data, outputConfig.includesNulls);
-
-		if ( !this.oldData.isEmpty() ) {
-			writeMapToJSON("old", this.oldData, true);
+		writeMapToJSON("data", this.data, outputConfig);
+		if( !this.oldData.isEmpty() ){
+			writeMapToJSON("old", this.oldData, outputConfig);
 		}
 
 		g.writeEndObject(); // end of row
 		g.flush();
 
-		return jsonFromStream();
+		if( outputConfig.encryptAll ) {
+			return encryptedJsonFromStream(outputConfig);
+		}
+		else{
+			return jsonFromStream();
+		}
+	}
+
+	private String encryptedJsonFromStream(MaxwellOutputConfig outputConfig){
```

Comment:
Doesn't look nice passing `outputConfig` everywhere just for the encryption keys. Encryptor should knows the keys IMO.